### PR TITLE
Add buffer arrays

### PR DESCRIPTION
**Pull request by csnewman**
**Connections**
https://github.com/gfx-rs/wgpu-rs/pull/850

**Description**
Adds buffer array bindings

**Testing**
TODO

### CODE REVIEW

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -1266,6 +1278,119 @@ impl<B: GfxBackend> Device<B> {
         })
     }
 
+    fn create_buffer_descriptor<'a>(
```
Comment (csnewman) [inline wgpu-core/src/device/mod.rs:1281]: This method has too many arguments causing a clippy error. Not sure the best way to clean this up.

Comment (kvark) [reply wgpu-core/src/device/mod.rs:1281]: in the worst case, just add the `[allow(...)]` on it

**Review by kvark — APPROVED** [review]
Looks reasonable to me!

**Review by cwfitzgerald — APPROVED** [review]

**Comment by kvark** [comment]
Let's make clippy happy and turn this into a non-draft PR

**Comment by kvark** [comment]
@csnewman can we finish this for 0.8?

**Comment by kvark** [comment]
Taking this over now

**Pull request merged**